src/main/scala/org/apache/spark/sql/hbase/HBaseSQLParser.scala

```diff
@@ -74,6 +80,34 @@ class HBaseSQLParser extends SqlParser {
         InsertValueIntoTableCommand(tableName, valueStringSeq)
     }
 
+  // Standard Syntax:
+  // UPDATE tablename SET column = value [, column = value ...] [WHERE expression]
+  protected lazy val update: Parser[LogicalPlan] =
+    (UPDATE ~> relation <~ SET) ~ rep1sep(updateColumn, ",") ~ (WHERE ~> expression) ^^ {
+      case table ~ updateColumns ~ exp =>
+        val (columns, values) = updateColumns.unzip
+        catalyst.logical.UpdateTable(
+          table.asInstanceOf[UnresolvedRelation].tableName,
+          columns.map(UnresolvedAttribute.quoted),
+          values,
+          Filter(exp, table))
+    }
+
+  protected lazy val updateColumn: Parser[(String, String)] =
+    ident ~ (EQ ~> ident) ^^ {
+      case column ~ value => (column, value)
+    }
+
+  // Standard Syntax:
+  // DELETE FROM tablename [WHERE expression]
+  protected lazy val delete: Parser[LogicalPlan] =
+    DELETE ~ FROM ~> relation ~ (WHERE ~> expression) ^^ {
+      case table ~ exp =>
+        catalyst.logical.DeleteFromTable(
+          table.asInstanceOf[UnresolvedRelation].tableName,
+          Filter(exp, table))
```

Comment:
@yzhou2001 for your point 1. i think it is ok now, because here i just do filter, that means i select all column from the table.